doc/doxygen-root/doxyfile

```diff
@@ -0,0 +1,2423 @@
+# Doxyfile 1.8.11
```

Comment:
1) Why is this a different version than the other files?
2) Should the existing doxyfile be removed?

Reply:
Updated doxyfile to version generated by doxygen 1.8.17 and removed src/doxyfile that was the original doxyfile and the basis for the new doxyfile.

Reply:
Turns out deleting src/doxyfile causes the run_doxygen check to fail, so I've restored it until that check is disabled.

Reply:
Could you perhaps add a commit that removes that file and adjusts the CI job that uses `run_doxygen` to do something that is meaningful with the new documentation set-up? The primary goal of the CI job is to make sure people don't add documentation that would cause new Doxygen warnings to appear.